core/unit_test/tstCommunicationPlan.hpp

```diff
@@ -0,0 +1,461 @@
+/****************************************************************************
+ * Copyright (c) 2018 by the Cabana authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cabana_CommunicationPlan.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <gtest/gtest.h>
+
+#include <mpi.h>
+
+#include <vector>
+#include <algorithm>
+
+namespace Test
+{
+//---------------------------------------------------------------------------//
+class CommPlanTester : public Cabana::CommunicationPlan<TEST_MEMSPACE>
+{
+  public:
+
+    CommPlanTester( MPI_Comm comm )
+        : Cabana::CommunicationPlan<TEST_MEMSPACE>(comm)
+    {}
+
+    template<class ViewType>
+    void createFromExportsAndNeighbors(
+        const ViewType& element_export_ranks,
+        const std::vector<int>& neighbor_ranks )
+    {
+        this->createFromExportsAndTopology(
+            element_export_ranks, neighbor_ranks );
+    }
+
+    template<class ViewType>
+    void createFromExports( const ViewType& element_export_ranks )
+    {
+        this->createFromExportsOnly( element_export_ranks );
+    }
+
+    template<class ViewType>
+    void createSteering( const ViewType& element_export_ranks )
+    {
+        this->createExportSteering( element_export_ranks );
+    }
+
+    template<class RankViewType, class IdViewType>
+    void createSteering(
+        const RankViewType& element_export_ranks,
+        const IdViewType& element_export_ids )
+    {
+        this->createExportSteering(
+            element_export_ranks, element_export_ids );
+    }
```

Comment:
yeah I had made this change so that they could be called in the unit test assuming they would be `protected`. Once I got on a GPU machine to test I realized they would need to be public and therefore hadn't changed this back to directly calling the object.